File: 프로그래머스/lv2/70129. 이진 변환 반복하기/이진 변환 반복하기.java
class Solution {
    public int[] solution(String s) {
        int[] answer = new int[2];
        int zero = 0;
        int count = 0;

        while(!s.equals("1")){
            int s_len = s.length();
            s = s.replace("0", "");
            zero += s_len-s.length();
            s = Integer.toBinaryString(s.length());
            count++;
        }

        
        answer[0] = count;
        answer[1] = zero;
        return answer;
    }
}

File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.js
function solution(s) {
    var answer = [0, 0];
    
    while (true) {
        if (s.length === 1) break;
        const re = s.replaceAll('0', '');
        answer[0]++;
        answer[1] += s.length - re.length;
        
        s = re.length.toString(2);
    }
    
    
    return answer;
}